sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py

```diff
@@ -43,6 +44,10 @@ def __init__(self, topic_hostname, credential, **kwargs):
         self._topic_hostname = topic_hostname
         auth_policy = _get_authentication_policy(credential)
         self._client = EventGridPublisherClientImpl(authentication_policy=auth_policy, **kwargs)
+        self._client._client._pipeline._impl_policies.append(
```

Comment:
That's not correct, when the pipeline is created there is a linked list of policies that is created with the `next` link, meaning this policy won't be called ever (this will be in the list, but the pipeline doesn't follow the list, it asks policies to call next):
https://github.com/Azure/azure-sdk-for-python/blob/eb4ed0ac018d1bf57c43d9c401722d43fd8bb1db/sdk/core/azure-core/azure/core/pipeline/_base.py#L71

Also, you're using four levels of _variables here :).

The right approach, as we discussed in scrum, is to define your own complete set of policies, since there is no API to insert a policy today